flaml/autogen/oai/completion.py

```diff
@@ -991,6 +999,21 @@ def extract_text(cls, response: dict) -> List[str]:
             return [choice["text"] for choice in choices]
         return [choice["message"].get("content", "") for choice in choices]
 
+    @classmethod
+    def extract_text_or_function_call(cls, response: dict) -> List[str]:
+        """Extract the text or function calls from a completion or chat response.
+
+        Args:
+            response (dict): The response from OpenAI API.
+
+        Returns:
+            A list of function calls in the responses.
+        """
+        choices = response["choices"]
+        if "text" in choices[0]:
+            return [choice["text"] for choice in choices]
```

Comment:
Is it possible that text and function calls exist in the same repsonse? (I kind of observed this, but need to double check.)

Reply:
"text" is for Completion response.
"message" is for ChatCompletion response.